federator/src/main.js

```diff
@@ -21,92 +21,93 @@ const StatusServer = require('./lib/Endpoints.js');
 StatusServer.init(logger);
 
 if(!config.mainchain || !config.sidechain) {
-    logger.error('Mainchain and Sidechain configuration are required');
-    process.exit();
+  logger.error('Mainchain and Sidechain configuration are required');
+  process.exit();
 }
 
 if (!config.etherscanApiKey) {
-    logger.error('Etherscan API configuration is required');
-    process.exit();
+  logger.error('Etherscan API configuration is required');
+  process.exit();
 }
 
 const heartbeat = new Heartbeat(config, log4js.getLogger('HEARTBEAT'));
 const mainFederator = new Federator(config, log4js.getLogger('MAIN-FEDERATOR'));
 const sideFederator = new Federator({
-    ...config,
-    mainchain: config.sidechain,
-    sidechain: config.mainchain,
-    storagePath: `${config.storagePath}/side-fed`
+  ...config,
+  mainchain: config.sidechain,
+  sidechain: config.mainchain,
+  storagePath: `${config.storagePath}/side-fed`
 }, log4js.getLogger('SIDE-FEDERATOR'));
 const mainFederatorNFT = new FederatorNFT.FederatorNFT({
   ...config,
   storagePath: `${config.storagePath}/nft`
 }, log4js.getLogger('MAIN-FEDERATOR'));
 const sideFederatorNFT = new FederatorNFT.FederatorNFT({
-    ...config,
-    mainchain: config.sidechain,
-    sidechain: config.mainchain,
-    storagePath: `${config.storagePath}/nft/side-fed`
+  ...config,
+  mainchain: config.sidechain,
+  sidechain: config.mainchain,
+  storagePath: `${config.storagePath}/nft/side-fed`
 }, log4js.getLogger('SIDE-FEDERATOR'));
 
 let pollingInterval = config.runEvery * 1000 * 60; // Minutes
 let scheduler = new Scheduler(pollingInterval, logger, { run: () => run() });
 
 scheduler.start().catch((err) => {
-    logger.error('Unhandled Error on start()', err);
+  logger.error('Unhandled Error on start()', err);
 });
 
 async function run() {
-    try {
-        await mainFederator.run();
-        await sideFederator.run();
-
-        if (config.mainchain.nftBridge == null) {
-          throw new CustomError('Main Federator NFT Bridge empty at config.mainchain.nftBridge');
-        }
-        await mainFederatorNFT.run();
-
-        if (config.sidechain.nftBridge == null) {
-          throw new CustomError('Side Federator NFT Bridge empty at config.sidechain.nftBridge');
-        }
-        await sideFederatorNFT.run();
-
-        await heartbeat.readLogs();
-    } catch(err) {
-        logger.error('Unhandled Error on run()', err);
-        process.exit();
+  try {
+    await mainFederator.run();

```

Comment:
do we want to also run a regular federator when running an NFT federator? I'm asking because with this boolean variable (`useNft`) we can run a regular federator or a regular federator + NFT federator, never an NFT federator only
@pmprete

Reply:
The idea is that the federator runs both process all the time.  Some chains may not have NFT available for them, so we want to avoid running the process in that case, also for testing purposes we want to enable/disable them.